src/Security/EventListener/DenyAccessListener.php

```diff
@@ -81,7 +81,17 @@ public function onKernelRequest(GetResponseEvent $event)
         $extraVariables['request'] = $request;
 
         if (!$this->resourceAccessChecker->isGranted($attributes['resource_class'], $isGranted, $extraVariables)) {
-            throw new AccessDeniedException();
+            $defaultMessage = 'Access Denied.';
+
+            if (isset($attributes['collection_operation_name'])) {
+                $message = $resourceMetadata->getCollectionOperationAttribute($attributes['collection_operation_name'], 'access_control_message', $defaultMessage, true);
+            } elseif (isset($attributes['item_operation_name'])) {
+                $message = $resourceMetadata->getItemOperationAttribute($attributes['item_operation_name'], 'access_control_message', $defaultMessage, true);
+            } else {
+                $message = $resourceMetadata->getCollectionOperationAttribute($attributes['subresource_operation_name'], 'access_control_message', $defaultMessage, true);
```

Comment:
In fact this should be `getSubresourceOperationAttribute`

Reply:
Updated.